File: medium/2149_rearrange_array_elements_by_sign/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> rearrangeArray(std::vector<int> &nums) {
    std::vector<int> rearrangedNums(nums.size());

    size_t positive = 0;
    size_t negative = 1;

    for (const int &num : nums) {
      if (num > 0) {
        rearrangedNums[positive] = num;
        positive += 2;
      } else {
        rearrangedNums[negative] = num;
        negative += 2;
      }
    }

    return rearrangedNums;
  }
};

File: medium/2149_rearrange_array_elements_by_sign/solution.py
class Solution:
    def rearrangeArray(self, nums: list[int]) -> list[int]:
        rearrangedNums = [0] * len(nums)

        positive = 0
        negative = 1

        for num in nums:
            if num > 0:
                rearrangedNums[positive] = num
                positive += 2
            else:
                rearrangedNums[negative] = num
                negative += 2

        return rearrangedNums
